### DISCUSSION TITLE
Gitlab Support for adding new projects to OpenUPM.com

### DISCUSSION THREAD

**Discussion by GDGRhinox** [openupm/openupm#2597] [Ideas]
Currently you can only add github repositories through the OpenUPM website:
![image](https://user-images.githubusercontent.com/80831813/142182433-2a1af128-1e78-4295-896e-09e4c2998c3e.png)

We host our open source projects on GitLab as well, some are mirrored through GitHub but it would be handier if we could just add  our GitLab projects directly.

**Comment by favoyang** [comment]
@GDGRhinox sorry that OpenUPM didn't support GitLab. We have the system coupled on GitHub API, so it gonna take quite some work to remove it to support raw git protocol.

As a workaround, you can create a mirror on GitHub with the help of a timer CI - https://github.com/marketplace/actions/git-sync-action (haven't tried that yet, but looks promising). Just make sure you sync everything including Git tags.

**Reply by GDGRhinox** [reply]
That's fair, we have some projects on GitHub as well, so mirroring or porting the repository shouldn't be too much of a challenge.
Might be interesting upgrade for the future, but I get that working with the API limits your capabilities at this time.
Thanks for the quick response
